package.json

```diff
@@ -83,6 +83,7 @@
     "react-cookie": "^6.1.1",
     "react-dom": "^18.2.0",
     "react-stately": "^3.28.0",
+    "server-only": "^0.0.1",
```

Comment:
I caved in and finally added this, as per Next.js's instructions: https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#keeping-server-only-code-out-of-the-client-environment